hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/ComplexAvroKeyGenerator.java

```diff
@@ -44,6 +47,11 @@ public ComplexAvroKeyGenerator(TypedProperties props) {
 
   @Override
   public String getRecordKey(GenericRecord record) {
+    if (doAutoGenerateRecordKeys()) {
```

Comment:
how about `if (doAutoGenerateRecordKeys() || getRecordKeyFields.isEmpty())`?

Reply:
nope. if not for auto record key generation, record key field cannot be empty. even if its, we should let KeyGenUtils.getRecordKey() handle it.